### PR TITLE
Automatically build the mod and send it to discord.

### PR DESCRIPTION
**Pull request by lenbot7**
Automatically build the mod and send it to discord.
Discord webhook is stored in the secret DISCORD_WEBHOOK.

### CODE REVIEW

**Comment by lenbot7** [comment]
You should change the secret to your own. It is stored in a secret so nobody can see it.

File: .github/workflows/build-public-discord.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        env:
+          DISCORD_WEBHOOK: ${{ secrets.DISCORD_WEBHOOK }}
+        with:
+          args: ./build/libs/*.*
+
+#            - name: Archive Artifacts
+#              uses: actions/upload-artifact@v3
+#              with:
+#                  name: createrobotics-${{ steps.get_mc_version.outputs.mc_version }}-nightly-${{ steps.get_hash.outputs.sha_short }}
+#                  path: build/libs/
```
Comment (solonovamax) [inline .github/workflows/build-public-discord.yml:81]: remove this.

File: .github/workflows/build-public-discord.yml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        with:
+          webhook: ${{ secrets.DISCORD_WEBHOOK }}
+          message-title: Building mod from branch ${{ steps.get_branch.outputs.branch }}
+
+      - name: Get MC version
+        uses: christian-draeger/read-properties@908f99d3334be3802ec7cb528395a69d19914e7b
+        id: get_mc_version
+        with:
+          path: gradle.properties
+          properties: mc_version
```
Comment (solonovamax) [inline .github/workflows/build-public-discord.yml:42]: not needed.

File: .github/workflows/build-public-discord.yml [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Get branch name
+        id: get_branch
+        run: echo "branch=${GITHUB_REF#refs/heads/}" >> $GITHUB_OUTPUT
+
+      - name: Get latest commit hash
+        id: get_hash
+        run: echo "sha_short=$(git rev-parse --short HEAD)" >> $GITHUB_OUTPUT
```
Comment (solonovamax) [inline .github/workflows/build-public-discord.yml:28]: also not needed.

File: .github/workflows/build-public-discord.yml [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+# documentation.
+# This workflow will build a Java project with Gradle and cache/restore any dependencies to improve the workflow execution time
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-gradle
+
+name: Java CI with Gradle
+
+on:
+  workflow_dispatch:
+  push:
+  pull_request:
```
Comment (solonovamax) [inline .github/workflows/build-public-discord.yml:13]: why is it running on PRs?

File: .github/workflows/build-public-discord.yml [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Get branch name
+        id: get_branch
+        run: echo "branch=${GITHUB_REF#refs/heads/}" >> $GITHUB_OUTPUT
```
Comment (solonovamax) [inline .github/workflows/build-public-discord.yml:24]: not needed.

File: .github/workflows/build-public-discord.yml [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+      - name: Get latest commit hash
+        id: get_hash
+        run: echo "sha_short=$(git rev-parse --short HEAD)" >> $GITHUB_OUTPUT
+
+      - name: Send building message to discord
+        if: github.event_name == 'push'
+        uses: realRobotix/action-discord-notifier@release-master
+        with:
+          webhook: ${{ secrets.DISCORD_WEBHOOK }}
+          message-title: Building mod from branch ${{ steps.get_branch.outputs.branch }}
```
Comment (solonovamax) [inline .github/workflows/build-public-discord.yml:35]: why do you need a message to say it's building? just publish one when the build is complete.

File: .github/workflows/build-public-discord.yml [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+        with:
+          arguments: build
+
+      # - name: Publish to GitHub Packages
+      #   if: github.event_name == 'push' && github.repository_id == '521749383'
+      #   uses: gradle/gradle-build-action@v2
+      #   with:
+      #     arguments: publishMavenJavaPublicationToGitHubPackagesRepository
+      #   env:
+      #     GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (solonovamax) [inline .github/workflows/build-public-discord.yml:67]: remove.

**Review by solonovamax — CHANGES_REQUESTED** [review]

File: .github/workflows/build-public-discord.yml (hunk 3 above)
Comment (lenbot7) [reply .github/workflows/build-public-discord.yml:28]: that *is* needed

Comment (solonovamax) [reply .github/workflows/build-public-discord.yml:28]: where? the only place it is used is commented out.

**Comment by kikugie** [comment]
I was said on the discord that its not needed, so might as well close

**Pull request closed without merging**